### ISSUE TITLE
Performance issue (v0.7.2) on Mac - high CPU utilisation and Energy consumption

### ISSUE THREAD

**Issue by delphym** [ramboxapp/community-edition#2509]
Just recently updated Rambox to it's latest available version 0.7.2 on MacOs v10.14.6 (18G1012)

Not exactly sure if it is specific to this new version, but that's when I noticed, when my Mac is in idle and its fans go crazy and its hot too.

Checking Apple's Activity Monitor it reveals that Rambox is on the top of the list of apps utilising the most of CPU (memory is not too bad IMHO), but also it is juicing a lot of its battery.

Pls, see attached screenshots
https://photos.app.goo.gl/dNisypEyoX34JzUX9

	
	-
	> Rambox 0.7.2
	> Electron 7.1.1
	> darwin x64 18.7.0

**Comment by delphym** [comment]
BTW: if it is going to help I'm using 6 services:
* Hangouts Chat
* LinkedIn
* Skype
* FB Messanger
* Hangouts
* Android messages

**Comment by delphym** [comment]
I've just updated to v0.7.3 ... by the 1st look it seems to have same performance issue.

I'd love to provide more info, if you tell me  (point me) what are you after...

**Comment by fvulich** [comment]
Could you try disabling one by one the services in order to detect which one is consuming an abnormal quantity of CPU cycles ?

**Comment by delphym** [comment]
Sorry for the late response. Over Xmas period I'm not much at my computer.
I'll try to do so most probably after 7th Jan.

**Comment by tobiascz** [comment]
I had a similar issue with one of my 2 Slack services. One of the services consumed a lot more recourses than the other one. An easy way of verifying is when you go to activity monitor (filter them by clicking process name tab) and you will see multiple Rambox Helper (Renderer) processes and exactly one Rambox Helper (GPU) process. Probably one of the Helper (Renderer) processes takes a lot of recourses. You can _Force quiet _ this exact process and check back at the Rambox guy which of your services you just killed (for me it was that one Slack service). 

Once I identified the problem causing service I checked the configuration of this service and found that I accidentally used in the url twice the _.slack_ name.
how it was: 
`https://#NAMEOFSLACKGROUP.slack.slack.com`  
how it should be:
`https://#NAMEOFSLACKGROUP.slack.com`  

I changed it and restarted and now it works just fine.

**Comment by fvulich** [comment]
@delphym Are you still experiencing this issue?

**Comment by delphym** [comment]
Hi @fvulich , so sorry for the late response.

Yes, I do and I tried to record a screencast of CPU/Energy impact when using Rambox with various services turned on & off.

Please watch it here if you could: https://drive.google.com/file/d/14D90D9o5BdfBGn58CyzJFE2dMV4L7BNO/view?usp=sharing

**Comment by parcae** [comment]
I have the same issue. However, I did not try to check which service it is coming from, but I just force-quited Rambox Pro.
I will try to do some more digging when it occurs again this week.

**Comment by surak** [comment]
Please, check #538 and #2820 .

TLDR: Close rambox and open from the terminal, like this:

 ```bash
/Applications/Rambox.app/Contents/MacOS/Rambox --disable-gpu
```

and see if it improves things.

**Comment by delphym** [comment]
Hi @surak , sorry for the delay.

Yes, indeed it helped a lot!!! Using the integrated GPU works great and decreased incredibly the amount of CPU and energy required by Rambox.

It doesn't feature in the top 10-20 list of most hungry apps anymore.
![image](https://user-images.githubusercontent.com/5643604/99899371-c782e080-2d0d-11eb-9327-70faa0a267e4.png)

**Comment by surak** [comment]
Yes, so the bug is still there, as there is no "normal" way to make it work as expected. I would keep the issue open until the Rambox people fix it properly (as in make the GUI do what it says it should do), do you agree?

**Comment by delphym** [comment]
You're actually very right. I got influence by seeing #538 being closed.

Question is, what is the Rambox developers' approach on this.  I hope they'll try to resolve it this time.
Thanks for stepping in.

**Comment by surak** [comment]
bump. The issue is still here. I save hours of battery by using the cpu instead of the gpu, with versions:

```
Version: 0.7.7
Platform: darwin (x64)
Electron: 7.2.4
Chromium: 78.0.3904.130
Node: 12.8.1```

Why is the gpu version so inefficient?
